mlflow/cli.py

```diff
@@ -453,14 +450,28 @@ def gc(backend_store_uri, run_ids):
         print("Run with ID %s has been permanently deleted." % str(run_id))
 
 
-cli.add_command(mlflow.models.cli.commands)
 cli.add_command(mlflow.deployments.cli.commands)
-cli.add_command(mlflow.sagemaker.cli.commands)
 cli.add_command(mlflow.experiments.commands)
 cli.add_command(mlflow.store.artifact.cli.commands)
-cli.add_command(mlflow.azureml.cli.commands)
 cli.add_command(mlflow.runs.commands)
 cli.add_command(mlflow.db.commands)
 
+try:
+    # pylint: disable=unused-import
+    import pandas  # noqa: E402
+    import numpy  # noqa: E402
+except ImportError as e:
+    # We are conditional loading these commands since the skinny client does
+    # not support them due to the pandas and numpy dependencies of MLflow Models
+    pass
+else:
```

Comment:
I'd rather we just try to import the CLI modules directly and catch `ImportError` (rather than try/catching import error on numpy and pandas preemptively). Our test coverage for the azureml / sagemaker / models CLIs should be sufficient to ensure that these CLIs are importable / usable when the full set of core dependencies is specified.